source/book/include/marlin/book/Condition.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+// -- std includes
+#include <functional>
+#include <optional>
+#include <regex>
+#include <string_view>
+#include <typeindex>
+
+// -- MarlinBook includes
+#include "marlin/book/EntryData.h"
+
+namespace marlin {
+  namespace book {
+    // -- MarlinBook forward declaration
+    class EntryKey ;
+
+    /**
+     *	@brief wrapper class for an Entry filter function.
+     */
+    class Condition {
+
+    public:
+      using FilterFn_t = std::function< bool( const EntryKey & ) > ;
+
+      /// default constructor.
+      Condition() ;
+
+      /// construct Condition from filter function.
+      Condition( const FilterFn_t &filterFn ) ;
+
+      /// copy constructor.
+      Condition( const Condition & ) = default ;
+
+      /**
+       *	@brief test Condition on key.
+       * 	@return true if key matches the condition.
+       */
+      bool operator()( const EntryKey &key ) const { return _fiterFn( key ); }
+
+      // TODO: 'and' and 'or' are keywords in c++(and = &&, or = ||)
```

Comment:
more an discuss, is Not, And, Or ok for the functions? because it's not the naming convention. When we overload the `&&` `||` operator `condition.and(condition2)` is valid ...